src/main/java/com/minecolonies/coremod/entity/ai/citizen/fisherman/EntityAIWorkFisherman.java

```diff
@@ -327,7 +327,8 @@ private IAIState findWater()
         executedRotations = 0;
         //If he can't find any pond, tell that to the player
         //If 20 ponds are already stored, take a random stored location
-        if (job.getPonds().size() >= MAX_PONDS)
+        //the fishman should not go to find water when fishman can find water but cant find pond
+        if (job.getPonds().size() >= MAX_PONDS||(lastPathResult!=null&&lastPathResult.pond==null&&job.getPonds().size()>0))
```

Comment:
Good point, can you adjust the formatting her (use whitespaces similar to the previous if)